geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -122,21 +123,26 @@ static IProcess groundSkyViewFactor() {
                 String baseName = "rsu_ground_sky_view_factor"
                 String outputTableName = prefixName + "_" + baseName
 
-                // Other local variables
-                // Size of the grid mesh used to sample each RSU (according to the point density, we take a factor 10
-                // in order to consider that for some LCZ(7), 90% of the surfaces may be covered by buildings and
-                // then will not be used for the SVF calculation)
-                def gms = Math.pow((1.0/pointDensity)/10,0.5)
-
                 // Create the needed index on input tables and the table that will contain the SVF calculation points
-                datasource.execute(("CREATE INDEX IF NOT EXISTS ids_inI ON $rsuTable($geometricColumnRsu) USING RTREE; "+
-                        "CREATE INDEX IF NOT EXISTS ids_inA ON $correlationBuildingTable($geometricColumnBu) USING RTREE; "+
-                        "CREATE INDEX IF NOT EXISTS id_inA ON $correlationBuildingTable($idColumnRsu); "+
-                        "DROP TABLE IF EXISTS $ptsRSUtot; CREATE TABLE $ptsRSUtot (pk serial, "+
+                H2gisSpatialTable rsuSpatialTable = datasource.getSpatialTable(rsuTable)
+                H2gisSpatialTable buildingSpatialTable = datasource.getSpatialTable(correlationBuildingTable)
+                if(!rsuSpatialTable[geometricColumnRsu].spatialIndexed){rsuSpatialTable[geometricColumnRsu].createSpatialIndex()}
+                if(!rsuSpatialTable[idColumnRsu].indexed){rsuSpatialTable[idColumnRsu].createIndex()}
+                if(!buildingSpatialTable[geometricColumnRsu].spatialIndexed){buildingSpatialTable[geometricColumnRsu].createSpatialIndex()}
+                if(!buildingSpatialTable[idColumnRsu].indexed){buildingSpatialTable[idColumnRsu].createIndex()}
+
+                def to_start = System.currentTimeMillis()
+
+                datasource.execute(("DROP TABLE IF EXISTS $ptsRSUtot; CREATE TABLE $ptsRSUtot (pk serial, "+
                         "the_geom geometry, id_rsu int)").toString())
 
                 // The points used for the SVF calculation should be selected within each RSU
                 datasource.eachRow("SELECT * FROM $rsuTable".toString()) { row ->
+                    // Size of the grid mesh used to sample each RSU (based on the building density + 10%) - if the
+                    // building density exceeds 90%, the LCZ 7 building density is then set to 90%
+                    def freeAreaDens = Math.max(1-(row[rsuBuildingDensityColumn]+0.1), 0.1)
+                    def gms = Math.pow(freeAreaDens/pointDensity,0.5)
+                    println(gms)
```

Comment:
instead of `println` you can use `LOGGER.info(String)` or `LOGGER.warn(String)`

Reply:
done